utilities/src/serialize/impls.rs

```diff
@@ -148,7 +148,46 @@ impl_canonical_serialization_uint!(u8);
 impl_canonical_serialization_uint!(u16);
 impl_canonical_serialization_uint!(u32);
 impl_canonical_serialization_uint!(u64);
-impl_canonical_serialization_uint!(usize);
+
+impl CanonicalSerialize for usize {
+    #[inline]
+    fn serialize_with_mode<W: Write>(&self, mut writer: W, _compress: Compress) -> Result<(), SerializationError> {
+        let u64_value = *self as u64;
```

Comment:
I'm unwrapping, but maybe we can add a conversion error variant to the serialization enum and use that instead

```suggestion
        let u64_value = u64::try_from(*self).unwrap();
```

Reply:
I think an `unwrap` is fine here, as extending to higher-bit architectures is likely to require additional work targeting `usize` (i.e. we'd prefer to break here rather than miss some other spot)